### PR TITLE
docs: various changes

### PR DESCRIPTION
**Pull request by andrii-bodnar**
# Description

- Remove the _Create React App_ article since it was almost duplicating the _React project_ article.
- Prettify the _React project_ article a bit.
- Hide the _TypeScript_ article from the sidebar (#1869). It probably will be completely removed in the future.
- Move the _Vue extractor_ article to the _API Reference_ section.
- Move the _ICU MessageFormat_ article to the _Guides_ section.
- Added the Babel<>SWC tabs switcher to the installation instructions.
- Sidebar reorder.

## Types of changes

[//]: # (What types of changes does your code introduce to Lingui?)
[//]: # (_Put an `x` in the boxes that apply_)

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Documentation update
- [ ] Examples update

## Checklist

[//]: # (_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._)

- [x] I have read the [CONTRIBUTING](https://github.com/lingui/js-lingui/blob/main/CONTRIBUTING.md) and [CODE_OF_CONDUCT](https://github.com/lingui/js-lingui/blob/main/CODE_OF_CONDUCT.md) docs
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added the necessary documentation (if appropriate)

### CODE REVIEW

File: website/docs/tutorials/setup-react.md [hunk 1]
```diff
@@ -37,82 +29,89 @@ This setup guide is for any project which uses React.
...
-        {
-          path: "<rootDir>/src/locales/{locale}/messages",
-          include: ["src"],
-        },
-      ],
-      format: "po",
-    };
-    ```
+:::info
+For those who prefer not to use Babel, Lingui offers the [SWC Plugin](/docs/ref/swc-plugin.md) as an alternative.
```
Comment (vonovak) [inline website/docs/tutorials/setup-react.md:35]: Many projects, such as those using newer versions of NextJS, do not use babel, and so I think we should have a "fork" all the way at the top that leads to separate installation instructions for SWC and for babel. Maybe using Tabs: https://docusaurus.io/docs/markdown-features/tabs)

I have recently set up a nextJS app with the pages router and RSC and I'd love to write docs for that. I'm not sure if that's going to be a separate guide (probably yes). Either way, I think we should warn people as soon as possible that installing babel macros doesn't make sense if you're using SWC.

File: website/docs/tutorials/setup-react.md [hunk 2]
```diff
@@ -37,82 +29,89 @@ This setup guide is for any project which uses React.
...
+│ en       │     0       │   0     │
+│ fr       │     0       │   0     │
+└──────────┴─────────────┴─────────┘
 
-## Further reading
+(use "lingui extract" to update catalogs with new messages)
+(use "lingui compile" to compile catalogs for production)
+```
+
+Congratulations! You've successfully set up project with Lingui. Now it's good time to follow [React tutorial](/docs/tutorials/react.md) or read about [ICU Message Format](/docs/guides/message-format.md) which is used in messages.
```
Comment (vonovak) [inline website/docs/tutorials/setup-react.md:106]:
```suggestion
Congratulations! You've successfully set up project with Lingui. Now it's good time to follow the [React tutorial](/docs/tutorials/react.md) or read about [ICU Message Format](/docs/guides/message-format.md) which is used in messages.
```

File: website/docs/tutorials/setup-vite.md [hunk 3]
```diff
@@ -159,4 +159,4 @@ See the [guide about dynamic loading catalogs](/docs/guides/dynamic-loading-cata
 
 See [Vite's official documentation](https://vitejs.dev/guide/features.html#dynamic-import) for more info about Vite dynamic imports.
 
-Congratulations! You've successfully set up a Vite project with LinguiJS. Now it's a good time to follow [React tutorial](/docs/tutorials/react.md) or read about [ICU Message Format](/docs/ref/message-format.md) which is used in messages.
+Congratulations! You've successfully set up a Vite project with LinguiJS. Now it's a good time to follow [React tutorial](/docs/tutorials/react.md) or read about [ICU Message Format](/docs/guides/message-format.md) which is used in messages.
```
Comment (vonovak) [inline website/docs/tutorials/setup-vite.md:162]:
```suggestion
Congratulations! You've successfully set up a Vite project with LinguiJS. Now it's a good time to follow the [React tutorial](/docs/tutorials/react.md) or read about [ICU Message Format](/docs/guides/message-format.md) which is used in messages.
```

File: website/docs/tutorials/setup-react.md (hunk 1 above)
Comment (andrii-bodnar) [reply website/docs/tutorials/setup-react.md:35]: @vonovak thank you for pointing, a separate Next.js article would be the best!

Comment (andrii-bodnar) [reply website/docs/tutorials/setup-react.md:35]: I was thinking about RSC support and probably we can document the current PoC `@lingui/react/server` approach until we have a clearer approach.

**Review by vonovak — APPROVED** [review]

File: website/docs/tutorials/setup-react.md (hunk 1 above)
Comment (thekip) [reply website/docs/tutorials/setup-react.md]: Would be really nice, to have that article. I also would like to warn people in that article  that using global `t` macro would be harmful in the hybrid applications such as NextJs. And thanks to new useLingui macro-hook now we have a much better alternative. I also think if we don't need a global instance of i18n configured, we could simplify a lot initialization in nextjs to something like that: https://github.com/lingui/js-lingui/issues/1339#issuecomment-1385470909

**Pull request merged**